ch/chproto/writer.go

```diff
@@ -178,7 +178,7 @@ func (w *Writer) DateTime(tm time.Time) {
 const secsInDay = 24 * 3600
 
 func (w *Writer) Date(tm time.Time) {
-	w.Uint16(uint16(unixTime(tm) / secsInDay))
+	w.Uint16(uint16((unixTime(tm) + TimeZoneOffset()) / secsInDay))
```

Comment:
I believe this is incorrect and instead the offset should be used when parsing the time received from clickhouse. You can check how https://github.com/ClickHouse/clickhouse-go handles this for details.